tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -264,35 +296,46 @@ private bool CacheInDevDrive(string existingPipCacheLocation)
 
     public void UpdateOptimizerListViewModelList()
     {
-        var cacheSubDir = "\\pip\\cache";
-        var environmentVariable = "PIP_CACHE_DIR";
-        var localAppDataDir = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
-        var existingPipCacheLocation = GetExistingCacheLocation(localAppDataDir.ToString(), cacheSubDir);
-        if (existingPipCacheLocation != null && !CacheInDevDrive(existingPipCacheLocation))
+        foreach (var cache in _cacheInfo)
         {
+            var existingCacheLocation = GetExistingCacheLocation(cache);
+            var environmentVariablePath = Environment.GetEnvironmentVariable(cache.EnvironmentVariable!);
+            if (environmentVariablePath is not null && CacheInDevDrive(environmentVariablePath!))
+            {
+                continue;
+            }

```

Comment:
Existing cache location seems to only check the cache directories not the environment variable for if someone set it elsewhere that isn't in the dev drive

Reply:
If the cache is in dev drive but the env variable is pointing to the wrong location, it is up to the developer to fix this. 

Currently with this check, if the env variable is set to non-devdrive location, we would prompt the same dialog again to move the cache where the cache is already on dev drive which is confusing and bad user experience. Instead in future we can have a different prompt to fix the env variable only.

Reply:
Ok for now I will remove

Reply:
Great, thanks! I will approve once you push the change.

Reply:
This is because you have not removed the env variable.

Reply:
Yeah, so we should keep the check for if the environment variable is already set.

If we don't then this will only work once

Reply:
> If the cache is in dev drive but the env variable is pointing to the wrong location, it is up to the developer to fix this.
> 
> Currently with this check, if the env variable is set to non-devdrive location, we would prompt the same dialog again to move the cache where the cache is already on dev drive which is confusing and bad user experience. Instead in future we can have a different prompt to fix the env variable only.

I can add to that check. If the environment path is set and Inside a dev drive. So that it doesn't break

Reply:
Don't worry, I am actively working on this, I will take your changes and build on top. Thanks for contributing.